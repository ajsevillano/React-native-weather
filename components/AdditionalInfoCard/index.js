//Libs
import { StyleSheet, Text, View } from 'react-native';
import { getTime } from '../../libs/time';
//SVG Icons
import Windy from '../../assets/additionalInfo/wind.svg';
import Sunrise from '../../assets/additionalInfo/sunrise.svg';
import Sunset from '../../assets/additionalInfo/sunset.svg';
import Pressure from '../../assets/additionalInfo/pressure.svg';
import Humidity from '../../assets/additionalInfo/humidity.svg';
import UV from '../../assets/additionalInfo/uv.svg';

const AdditionalInfoCard = ({ current }) => {
  return (
    <View style={styles.additionalInfo}>
      <Text style={styles.moreInfo}>Additional info</Text>
      <View style={styles.infoBlock}>
        <View style={styles.InfoContainerLeft}>
          <Windy />
          <Text style={styles.textWind}>
            Wind:{' '}
            {!current ? (
              '- - m/h'
            ) : (
              <Text style={styles.boldData}>{`${current.wind_speed}m/h`}</Text>
            )}
          </Text>
        </View>
        <View style={styles.InfoContainerRight}>
          <Humidity />
          <Text style={styles.textWind}>
            Humidity:{' '}
            {!current ? (
              '- -%'
            ) : (
              <Text style={styles.boldData}>{`${current.humidity}%`}</Text>
            )}
          </Text>
        </View>
      </View>
      <View style={styles.infoBlock}>
        <View style={styles.InfoContainerLeft}>
          <Sunrise />
          <Text style={styles.textWind}>
            Sunrise:{' '}
            {!current ? (
              '--:--'
            ) : (
              <Text style={styles.boldData}>{getTime(current.sunrise)}</Text>
            )}
          </Text>
        </View>
        <View style={styles.InfoContainerRight}>
          <Sunset />
          <Text style={styles.textWind}>
            Sunset:{' '}
            {!current ? (
              '--:--'
            ) : (
              <Text style={styles.boldData}>{getTime(current.sunset)}</Text>
            )}
          </Text>
        </View>
      </View>
      <View style={styles.infoBlock}>
        <View style={styles.InfoContainerLeft}>
          <Pressure />
          <Text style={styles.textWind}>
            Press.:{' '}
            {!current ? (
              '- -  mb'
            ) : (
              <Text style={styles.boldData}>{`${current.pressure} mb`}</Text>
            )}
          </Text>
        </View>
        <View style={styles.InfoContainerRight}>
          <UV />
          <Text style={styles.textWind}>
            UV:{' '}
            {!current ? (
              '- -'
            ) : (
              <Text style={styles.boldData}> {`${current.uvi}`}</Text>
            )}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  additionalInfo: {
    flex: 1,
    backgroundColor: 'white',
    paddingLeft: 40,
    paddingRight: 40,
    paddingBottom: 30,
  },

  InfoContainerLeft: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
  },

  InfoContainerRight: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginLeft: 35,
  },

  textWind: {
    fontSize: 17,
    color: '#777575',
    marginLeft: 5,
  },

  boldData: {
    fontWeight: '500',
    color: '#273365',
  },

  moreInfo: {
    fontSize: 20,
    marginTop: 30,
    marginBottom: 10,
    borderRadius: 10,
    fontWeight: 'bold',
    color: '#273365',
  },

  humidity: {
    fontSize: 17,
    color: '#777575',
  },

  infoBlock: {
    flex: 1,
    flexDirection: 'row',
    marginTop: 10,
    alignItems: 'flex-end',
  },
});

export default AdditionalInfoCard;
