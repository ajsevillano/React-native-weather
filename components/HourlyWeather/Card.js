//Libs
import { Text, View } from 'react-native';
import getIcons from '../../libs/getIcons';

//Context
import { useContext } from 'react';
import ThemeContext from '../../context/theme';

//Styles
import styles from './Card.styles';

const Card = ({ item }) => {
  const time = item?.time.split(' ')[1];
  const { theme } = useContext(ThemeContext);
  return (
    <View style={[styles.hourlyWeatherCard, { backgroundColor: '#f5f5f5' }]}>
      <Text style={[styles.temperatureText, { color: theme.text }]}>
        {item?.temperature.toFixed(0)}Â°
      </Text>
      {getIcons(item?.iconCode, item?.isDay, 'small')}
      <Text style={[styles.hour, { color: theme.text }]}>{time}</Text>
    </View>
  );
};

export default Card;
