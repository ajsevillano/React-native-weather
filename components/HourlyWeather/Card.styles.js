import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  hourlyWeatherCard: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 18,
    paddingVertical: 10,
    borderRadius: 10,
    marginRight: 20,
  },

  light_background: {
    backgroundColor: '#f5f5f5',
  },

  dark_background: {
    backgroundColor: '#222222',
  },

  temperatureText: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 15,
  },

  light_text: {
    color: '#273365',
  },

  dark_text: {
    color: 'white',
  },

  hour: {
    marginTop: 15,
  },

  now: {
    fontWeight: '700',
  },
});

export default styles;
