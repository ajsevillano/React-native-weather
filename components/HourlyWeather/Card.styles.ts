import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  hourlyWeatherCard: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 18,
    paddingVertical: 10,
    borderRadius: 10,
    marginRight: 20,
  },

  temperatureText: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 15,
  },

  hour: {
    marginTop: 15,
  },

  now: {
    fontWeight: '700',
  },
});

export default styles;
