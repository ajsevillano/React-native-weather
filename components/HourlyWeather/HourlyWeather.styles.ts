import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  Container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    paddingLeft: 40,
    paddingRight: 40,
  },

  titleHeader: {
    fontSize: 20,
    marginBottom: 20,
    fontWeight: 'bold',
    color: '#273365',
  },

  //Theme
  light_background: {
    backgroundColor: 'white',
  },
  dark_background: {
    backgroundColor: '#1b1b1b',
  },

  light_text: {
    color: '#273365',
  },
  dark_text: {
    color: 'white',
  },
});

export default styles;
