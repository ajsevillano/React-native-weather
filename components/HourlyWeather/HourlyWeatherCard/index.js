//Libs
import { StyleSheet, Text, View, FlatList } from 'react-native';
import { getTime } from '../../../libs/time';
import { getWeatherIcon } from '../../../libs/getIcons';

const HourlyWeatherCard = ({ item }) => {
  return (
    <View style={styles.hourlyWeatherCard}>
      <Text style={styles.temperatureText}>{item.temp.toFixed(0)}Â°</Text>
      {getWeatherIcon(item.weather, 'small')}
      <Text style={styles.hour}>{getTime(item.dt)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  hourlyWeatherCard: {
    // flexDirection: 'column',
    alignItems: 'center',
    // paddingLeft: 18,
    // paddingRight: 18,
    // paddingBottom: 16,
    // paddingTop: 16,
    borderRadius: 10,
    backgroundColor: '#f5f5f5',
    marginRight: 20,
  },

  temperatureText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#273365',
    marginBottom: 15,
  },

  hour: {
    marginTop: 15,
    color: '#273365',
  },
});

export default HourlyWeatherCard;
