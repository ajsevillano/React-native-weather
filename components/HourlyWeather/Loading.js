//Libs
import { Text, View, ActivityIndicator } from 'react-native';

//Context
import { useContext } from 'react';
import ThemeContext from '../../context/theme';

// Styles
import styles from './Loading.styles';

const Loading = () => {
  //Theme from Context
  const theme = useContext(ThemeContext);
  return (
    <>
      <View style={styles.Container}>
        <Text style={[styles.loadingText, { color: theme.text }]}>Loading</Text>
        <ActivityIndicator size='large' color={theme === 'light' ? '#273365' : 'white'} />
      </View>
    </>
  );
};

export default Loading;
