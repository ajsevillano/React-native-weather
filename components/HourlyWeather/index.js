//Libs
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  ActivityIndicator,
} from 'react-native';
//Components
import HourlyWeatherCard from './HourlyWeatherCard';

const HourlyWeather = ({ hourly }) => {
  /* Filtering the hourly array to only show the next 8 hours. */
  const filterHours =
    hourly && hourly.filter((hour, index) => index > 0 && index < 8);

  return (
    <View style={styles.hourlySectionContainer}>
      <View>
        <Text style={styles.titleHeader}>Hourly weather</Text>
      </View>
      {!hourly ? (
        <>
          <View style={styles.loadingIndicatorContainer}>
            <Text style={styles.loadingText}>Loading</Text>
            <ActivityIndicator size="large" color="#273365" />
          </View>
        </>
      ) : (
        <View>
          <FlatList
            showsHorizontalScrollIndicator={false}
            horizontal={true}
            data={filterHours}
            renderItem={({ item }) => <HourlyWeatherCard item={item} />}
          ></FlatList>
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  hourlySectionContainer: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: 'white',
    justifyContent: 'center',
    paddingLeft: 40,
    paddingRight: 40,
  },

  titleHeader: {
    fontSize: 20,
    marginBottom: 20,
    fontWeight: 'bold',
    color: '#273365',
  },

  loadingIndicatorContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: 20,
    paddingTop: 20,
  },

  loadingText: {
    marginBottom: 20,
  },
});

export default HourlyWeather;
