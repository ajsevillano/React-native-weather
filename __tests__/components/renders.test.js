//Libs
import renderer from 'react-test-renderer';

//Components
import AdditionalInfoCard from '../../components/AdditionalInfoCard';
import HourlyWeather from '../../components/HourlyWeather';
import HourlyWeatherCard from '../../components/HourlyWeather/HourlyWeatherCard';
import WeatherCard from '../../components/WeatherCard';

describe('Test all components render properly', () => {
  it(`Additional info card renders correctly`, () => {
    const tree = renderer.create(<AdditionalInfoCard></AdditionalInfoCard>);
    expect(tree).toMatchSnapshot();
  });
  it(`Hourly Weather component renders correctly`, () => {
    const tree = renderer.create(<HourlyWeather></HourlyWeather>);
    expect(tree).toMatchSnapshot();
  });
  it(`Hourly Weather card component renders correctly`, () => {
    const tree = renderer.create(<HourlyWeatherCard></HourlyWeatherCard>);
    expect(tree).toMatchSnapshot();
  });
  it(`Weather card renders correctly`, () => {
    const tree = renderer.create(<WeatherCard></WeatherCard>);
    expect(tree).toMatchSnapshot();
  });
});
