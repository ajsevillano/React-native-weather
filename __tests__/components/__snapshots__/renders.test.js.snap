// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all components render properly Additional info component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "flex": 1.5,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingLeft": 40,
        "paddingRight": 20,
      },
      Object {
        "backgroundColor": "#1b1b1b",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 5,
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    Additional info
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-wind"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#d3d9ff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4292073983}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M5 8h8.5a2.5 2.5 0 1 0-2.34-3.24M3 12h15.5a2.5 2.5 0 1 1-2.34 3.24M4 16h5.5a2.5 2.5 0 1 1-2.34 3.24"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        Wind
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          undefined m/h
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-droplet"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#d3d9ff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4292073983}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M6.8 11a6 6 0 1 0 10.396 0l-5.197-8-5.2 8z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        Humidity
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          undefined %
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-sunrise"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#ffbf00"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4294950656}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M3 17h1m16 0h1M5.6 10.6l.7.7m12.1-.7-.7.7M8 17a4 4 0 0 1 8 0M3 21h18M12 9V3l3 3M9 6l3-3"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        Sunrise
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          ERROR 
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-sunset"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#ff9300"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4294939392}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M3 17h1m16 0h1M5.6 10.6l.7.7m12.1-.7-.7.7M8 17a4 4 0 0 1 8 0M3 21h18M12 3v6l3-3M9 6l3 3"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        Sunset
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          ERROR 
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-temperature"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#d3d9ff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4292073983}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M10 13.5a4 4 0 1 0 4 0V5a2 2 0 0 0-4 0v8.5M10 9h4"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        Press
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          undefined Mb
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingVertical": 8,
          "width": "50%",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        className="icon icon-tabler icon-tabler-eyeglass"
        fill="none"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#d3d9ff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.5}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={4292073983}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={1.5}
        >
          <RNSVGPath
            d="M0 0h24v24H0z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={null}
          />
          <RNSVGPath
            d="M8 4H6L3 14M16 4h2l3 10M10 16h4M21 16.5a3.5 3.5 0 0 1-7 0V14h7v2.5M10 16.5a3.5 3.5 0 0 1-7 0V14h7v2.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#777575",
            "fontSize": 17,
            "marginLeft": 5,
          }
        }
      >
        UV
        :
         
        <Text
          style={
            Array [
              Object {
                "fontWeight": "500",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          undefined 
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test all components render properly Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffbf00",
      "borderRadius": 10,
      "flexDirection": "row",
      "marginTop": 20,
      "opacity": 1,
      "paddingHorizontal": 70,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#273365",
        "fontSize": 25,
        "fontWeight": "700",
      }
    }
  />
</View>
`;

exports[`Test all components render properly Hourly Weather component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "paddingLeft": 40,
        "paddingRight": 40,
      },
      Object {
        "backgroundColor": "#1b1b1b",
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#273365",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 20,
          },
          Object {
            "color": "white",
          },
        ]
      }
    >
      Hourly weather
    </Text>
  </View>
  <View>
    <RCTScrollView
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`Test all components render properly Weather card renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#222222",
      },
    ]
  }
/>
`;
